dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -325,7 +346,7 @@ public FunctionView Describe()
 
```

Comment:
AddVariablesToContext(this.State, functionContext);
   - I *think* this can be moved above the loop. Looks like this has to only be called once before the first run, then the UpdateContextWithOutputs takes care of the remainder.

Reply:
Not a blocker. Filed a task to check this out.

Reply:
Yeah sounds good, will want to make sure I test this change well, for follow up.